File: LeetCode/338.counting-bits/counting-bits.java
class Solution {
    public int[] countBits(int num) {
        int[] dp = new int[num + 1];
        dp[0] = 0;
        for (int i = 1; i <= num; i++){
            dp[i] = dp[i >> 1] + (i & 1);
        }

        return dp;
    }
}

File: LeetCode/338.counting-bits/counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int num) {
        vector<int>res = vector<int>(num + 1, 0);
        int cnt = 0;
        for (int i = 1; i <= num; i++){
            res[i] = res[i >> 1] + (i & 1);
        }

        return res;
    }
};